specification/azurefleet/AzureFleet.Management/baseVirtualMachineProfile.tsp

```diff
@@ -0,0 +1,2524 @@
+   */
+  configurationReference?: string;
+
+  /**
+   * Optional, If true, any failure for any operation in the VmApplication will fail
+   * the deployment
+   */
+  treatFailureAsDeploymentFailure?: boolean;
+
+  /**
+   * If set to true, when a new Gallery Application version is available in PIR/SIG,
+   * it will be automatically updated for the VM/VMSS
+   */
+  enableAutomaticUpgrade?: boolean;
+}
+
+/**
+ * Specifies the hardware settings for the virtual machine scale set.
+ */
+model VirtualMachineScaleSetHardwareProfile {
+  /**
+   * Specifies the properties for customizing the size of the virtual machine.
+   * Minimum api-version: 2021-11-01. Please follow the instructions in [VM
+   * Customization](https://aka.ms/vmcustomization) for more details.
+   */
+  vmSizeProperties?: VMSizeProperties;
+}
+
+/**
+ * Specifies VM Size Property settings on the virtual machine.
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" ""
+model VMSizeProperties {
+  /**
+   * Specifies the number of vCPUs available for the VM. When this property is not
+   * specified in the request body the default behavior is to set it to the value of
+   * vCPUs available for that VM size exposed in api response of [List all available
+   * virtual machine sizes in a
+   * region](https://docs.microsoft.com/en-us/rest/api/compute/resource-skus/list).
+   */
+  vCPUsAvailable?: int32;
+
+  /**
+   * Specifies the vCPU to physical core ratio. When this property is not specified
+   * in the request body the default behavior is set to the value of vCPUsPerCore
+   * for the VM Size exposed in api response of [List all available virtual machine
+   * sizes in a
+   * region](https://docs.microsoft.com/en-us/rest/api/compute/resource-skus/list).
+   * **Setting this property to 1 also means that hyper-threading is disabled.**
+   */
+  vCPUsPerCore?: int32;
+}
+
+/**
+ * Specifies the service artifact reference id used to set same image version for
+ * all virtual machines in the scale set when using 'latest' image version.
+ * Minimum api-version: 2022-11-01
+ */
+model ServiceArtifactReference {
+  /**
+   * The service artifact reference id in the form of
+   * /subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Compute/galleries/{galleryName}/serviceArtifacts/{serviceArtifactName}/vmArtifactsProfiles/{vmArtifactsProfilesName}
+   */
+  id?: armResourceIdentifier;
+}
+
+/**
+ * Specifies the security posture to be used for all virtual machines in the scale
+ * set. Minimum api-version: 2023-03-01
+ */
+model SecurityPostureReference {
+  /**
+   * The security posture reference id in the form of
+   * /CommunityGalleries/{communityGalleryName}/securityPostures/{securityPostureName}/versions/{major.minor.patch}|{major.*}|latest
+   */
+  id?: string;
+
+  /**
+   * List of virtual machine extensions to exclude when applying the Security
+   * Posture.
+   */
+  excludeExtensions?: VirtualMachineExtension[];
```

Comment:
Just a heads-up. It may affect your service.

Note that there is recent break in CRP on this property.
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-03-01/virtualMachineScaleSet.json#L6345-L6351

I do now know why the break. Just found this during upgrade of compute SDK...

Reply:
Oh, so we should also update to the latest and change its type to string[]?

Reply:
It may depend on the CRP api-version your service uses for this response (and what the response really is).

PS: here is CRP's response about why this break
https://github.com/Azure/azure-rest-api-specs/pull/28891#issuecomment-2140744373

Reply:
Thanks. We are using a range of CRP versions. Is there a way to define a different model in typespec when a particular [version ](https://github.com/Azure/azure-rest-api-specs/blob/12a9609361d21154a0fb141dd682ba3f2a437d7f/specification/azurefleet/AzureFleet.Management/fleet.tsp#L102) is passed?

Reply:
Updated